core/test/integration/model/model_posts_spec.js

```diff
@@ -475,12 +442,11 @@ describe('Post Model', function () {
                 }).catch(function (err) {
                     should.exist(err);
                     (err instanceof errors.ValidationError).should.eql(true);
-                    done();
                 });
             });
 
-            it('draft -> scheduled: invalid published_at update', function (done) {
-                PostModel.findOne({status: 'draft'}).then(function (results) {
+            it('draft -> scheduled: invalid published_at update', function () {
+                return PostModel.findOne({status: 'draft'}).then(function (results) {
```

Comment:
Actually on second thought, the approach you used below _should_ work. Instead of using `done`, just throw an error in the `then` method. Regardless, this particular test needs to have another `then` check in there.